protostar/starknet/cheatable_syscall_handler.py

```diff
@@ -23,6 +25,14 @@ def __init__(self, message: str):
 
 
 class CheatableSysCallHandler(BusinessLogicSysCallHandler):
+    def __init__(self, state: SyncState, **kwargs):
+        # This field must be set before entering super constructor,
+        # because it calls the setter for the `block_info` property.
```

Comment:
Why does it matter when this propery is set?

Reply:
Because of this https://github.com/starkware-libs/cairo-lang/blob/v0.10.0/src/starkware/starknet/core/os/syscall_utils.py#L77-L80, which is a super super constructor of this class.